controllers/pwas/crud.js

```diff
@@ -58,37 +61,78 @@ router.get('/add', (req, res) => {
  */
 // [START add]
 router.post('/add', (req, res, next) => {
-  const data = req.body;
+  const manifestUrl = req.body.manifestUrl;
+  const idToken = req.body.idToken;
 
-  const callback = (err, savedData) => {
-    if (err) {
-      if (typeof err === 'number') {
-        switch (err) {
-          case pwaModel.E_ALREADY_EXISTS:
-            res.render('pwas/form.hbs', {
-              pwa: {
-                manifestUrl: data.manifestUrl
-              },
-              error: 'manifest already exists'
-            });
-            return;
-          case pwaModel.E_MANIFEST_ERROR:
-            res.render('pwas/form.hbs', {
-              pwa: {
-                manifestUrl: data.manifestUrl
-              },
-              error: 'error loading manifest' // could be 404, not JSON, domain does not exist
-            });
-            return;
-          default:
-        }
-      }
-      return next(err);
-    }
-    res.redirect(req.baseUrl + '/' + savedData.id);
-  };
+  if (!manifestUrl) {
+    res.render('pwas/form.hbs', {
+      pwa: {
+        manifestUrl: ''
+      },
+      error: 'no manifest provided'
+    });
+    return;
+  }
 
-  pwaModel.save(data, callback);
+  if (!idToken) {
+    res.render('pwas/form.hbs', {
+      pwa: {
+        manifestUrl: manifestUrl
+      },
+      error: 'user not logged in'
+    });
+    return;
+  }
+
+  verifyIdToken(CLIENT_ID, CLIENT_SECRET, idToken)
+    .then(user => {
+      const pwa = {
+        manifestUrl: manifestUrl,
+        createdOn: new Date(),
+        updatedOn: new Date(),
+        user: {
+          id: user.getPayload().sub,
+          email: user.getPayload().email
+        }
```

Comment:
@juliantoledo Do we want to store the email? I think in the absence of a good use for it, we should avoid saving it. Might be also a good idea to HMAC the id as well, especially if it's not obvious that a submission is linked to a user.